_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -102,4 +102,130 @@ test.describe('Snapshot Repositories', () => {
+        'https://fedorapeople.org/groups/katello/fakerepos/zoo3',
+        'https://fedorapeople.org/groups/katello/fakerepos/zoo4',
+      ];
+      for (const url of zooUrls) {
+        await deleteAllRepos(page, `&url=${url}`);
+      }
+    });
+
+    await test.step('Create a repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByLabel('URL').fill('https://fedorapeople.org/groups/katello/fakerepos/zoo/');
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Edit the repository', async () => {
+      for (let i = 2; i <= 4; i++) {
+        const row = await getRowByNameOrUrl(page, repoName);
+        await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+        await test.step(`Edit repository and create snapshot ${i}`, async () => {
+          // Open the edit modal
+          await row.getByLabel('Kebab toggle').click();
+          await row.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });
+          await page
+            .getByLabel('URL')
+            .fill(`https://fedorapeople.org/groups/katello/fakerepos/zoo${i}/`);
+          await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+          await expect(row.getByText('Valid')).toBeVisible({ timeout: 70000 });
+        });
+      }
+    });
+
+    await test.step('Create a template', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      await row.getByRole('button', { name: 'Kebab toggle' }).click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      // pause to allow the snapshots to load
+      await page.waitForTimeout(2000);
+      const snapshotCount =
+        (await page.getByTestId('snapshot_list_table').locator('tr').count()) - 1; // Subtract 1 for the header row
+      expect(snapshotCount).toBe(4);
+      await navigateToTemplates(page);
+      await page.getByRole('button', { name: 'Add content template' }).click();
+      await page.getByRole('button', { name: 'Select architecture' }).click();
+      await page.getByRole('option', { name: 'aarch64' }).click();
+      await page.getByRole('button', { name: 'Select version' }).click();
+      await page.getByRole('option', { name: 'el9' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
```

Comment:
This failed in CI. I did not investigate a lot, but my first thought was - You click on a button and then want to click on it on the next screen. But the test is quick, so it will probably click on the same button twice - before it gets to the other screen. You should put there some wait for the next screen to be visible before clicking next time.

Reply:
This approach is working well for me now.

But i think you're right, it could lead to issues in the future.
So I'm modifying this line and adding an assertion to ensure the model is present for both this and the next page models.

Thanks!